cms/users/tests/test_viewsets.py

```diff
@@ -0,0 +1,58 @@
+from django.test import TestCase
+from django.urls import reverse
+from wagtail.test.utils import WagtailTestUtils
+
+from cms.users.tests.factories import UserFactory
+from cms.users.viewsets import user_chooser_viewset
+
+
+class TestUserChooserViewSet(WagtailTestUtils, TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.chooser_url = user_chooser_viewset.widget_class().get_chooser_modal_url()
+        cls.chooser_results_url = reverse(user_chooser_viewset.get_url_name("choose_results"))
+
+        cls.user_without_name = UserFactory(username="noname", first_name="", last_name="", is_active=False)
+        cls.user_with_first_name = UserFactory(username="first_name_user", first_name="Data", last_name="")
+        cls.user_with_last_name = UserFactory(username="last_name_user", first_name="", last_name="La Forge ")
+        cls.user_with_full_name = UserFactory(
+            username="full_name_user", first_name="Jean-Luc", last_name="Picard", is_superuser=True
+        )
+
+    def setUp(self):
+        self.login(user=self.user_with_full_name)
+
+    def test_chooser_viewset(self):
+        response = self.client.get(self.chooser_url)
+
+        self.assertContains(response, self.user_without_name.username, 2)  # Name and username
+        self.assertContains(response, self.user_with_first_name.username)
```

Comment:
I think the comments should be on the line below, as `user_with_first_name` has both name and username, while `user_without_name` just has the username and no name. Unless I misunderstood the comment.
```suggestion
        self.assertContains(response, self.user_without_name.username, 2)
        self.assertContains(response, self.user_with_first_name.username)  # Name and username
```

Reply:
not quite. if the user doesn't have a fullname `get_user_display_name` defaults to the username - https://github.com/wagtail/wagtail/blob/bfcd0c7496279137c28ebd00a304fa2a514df47b/wagtail/admin/utils.py#L52, so this checks that the username is displayed twice

Reply:
Makes sense, thanks!